File: backend/app/schemas/report.py
from datetime import date
from uuid import UUID

from pydantic import BaseModel


class DailyReport(BaseModel):
    date: date
    sessions_count: int
    problems_attempted: int
    problems_correct: int
    xp_earned: int
    time_spent_minutes: int
    knowledge_points_reviewed: list[UUID]


class WeeklyReport(BaseModel):
    week_start: date
    week_end: date
    total_sessions: int
    total_problems: int
    total_correct: int
    total_xp: int
    total_time_minutes: int
    streak_days: int
    mastery_changes: dict[str, float]

File: frontend/src/types/report.ts
// Types derived from backend Pydantic schemas:
//   backend/app/schemas/report.py

/** Backend: DailyReport – GET /reports/daily */
export interface DailyReport {
  date: string;
  sessions_count: number;
  problems_attempted: number;
  problems_correct: number;
  xp_earned: number;
  time_spent_minutes: number;
  knowledge_points_reviewed: string[];
}

/** Backend: WeeklyReport – GET /reports/weekly */
export interface WeeklyReport {
  week_start: string;
  week_end: string;
  total_sessions: number;
  total_problems: number;
  total_correct: number;
  total_xp: number;
  total_time_minutes: number;
  streak_days: number;
  mastery_changes: Record<string, number>;
}
